### PR TITLE
Fixed NATS volume mounts

### LINKED ISSUE #3756: [Nats]: WithJetStream method takes a mount path that assumes a bind mount
https://github.com/dotnet/aspire/blob/4829f56034c6ea799743c0550cdbd6c3d64e3ac5/src/Aspire.Hosting.Nats/NatsBuilderExtensions.cs#L36-L47

Seems like we should remove the path from this method and instead follow the pattern we've established for containers that have well known directories for persistence. 

e.g.

https://github.com/dotnet/aspire/blob/4829f56034c6ea799743c0550cdbd6c3d64e3ac5/src/Aspire.Hosting.Kafka/KafkaBuilderExtensions.cs#L40-L51

cc @mtmk

### PR DESCRIPTION
**Pull request by mtmk**
* resolves #3756 
 ###### Microsoft Reviewers: [Open in CodeFlow](https://microsoft.github.io/open-pr/?codeflow=https://github.com/dotnet/aspire/pull/3763)

### CODE REVIEW

**Comment by mitchdenny** [comment]
I did one too #3767.

I know very little about NATS though. I'm assuming that the -js and -sd arguments can be taken in isolation (although I guess -js doesn't make much sense without -sd?).

**Comment by davidfowl** [comment]
😄

File: src/Aspire.Hosting.Nats/NatsBuilderExtensions.cs [hunk 1]
```diff
@@ -30,19 +31,28 @@ public static IResourceBuilder<NatsServerResource> AddNats(this IDistributedAppl
...
-        if (srcMountPath != null)
-        {
-            args.Add("-sd");
-            args.Add("/data");
-            builder.WithBindMount(srcMountPath, "/data");
-        }
+    /// <param name="builder">The resource builder.</param>
+    /// <returns>The <see cref="IResourceBuilder{T}"/>.</returns>
+    public static IResourceBuilder<NatsServerResource> WithJetStream(this IResourceBuilder<NatsServerResource> builder)
+        => builder.WithArgs("-js", "-sd", "/var/lib/nats");
```
Comment (davidfowl) [inline src/Aspire.Hosting.Nats/NatsBuilderExtensions.cs:37]: What happens if you call this but you don't call `WithDataVolume`

File: src/Aspire.Hosting.Nats/NatsBuilderExtensions.cs [hunk 2]
```diff
@@ -30,19 +31,28 @@ public static IResourceBuilder<NatsServerResource> AddNats(this IDistributedAppl
     /// <summary>
     /// Adds JetStream support to the NATS server resource.
     /// </summary>
-    /// <param name="builder">NATS resource builder.</param>
-    /// <param name="srcMountPath">Optional mount path providing persistence between restarts.</param>
-    /// <returns>A reference to the <see cref="IResourceBuilder{T}"/>.</returns>
-    public static IResourceBuilder<NatsServerResource> WithJetStream(this IResourceBuilder<NatsServerResource> builder, string? srcMountPath = null)
```
Comment (eerhardt) [inline src/Aspire.Hosting.Nats/NatsBuilderExtensions.cs:36]: What's the policy on making API breaking changes right now? Is this supposed to be left in and `[Obsolete]`?

File: src/Aspire.Hosting.Nats/NatsBuilderExtensions.cs (hunk 1 above)
Comment (eerhardt) [reply src/Aspire.Hosting.Nats/NatsBuilderExtensions.cs:37]: Seems like a good test to add.

**Comment by davidfowl** [comment]
I think we should keep this one @mitchdenny can close the other one.

File: src/Aspire.Hosting.Nats/NatsBuilderExtensions.cs (hunk 1 above)
Comment (mtmk) [reply src/Aspire.Hosting.Nats/NatsBuilderExtensions.cs:37]: I think data just stays on the ephemeral container storage. Could you give me some ideas about the test. Happy to add them too.

File: src/Aspire.Hosting.Nats/NatsBuilderExtensions.cs (hunk 2 above)
Comment (mtmk) [reply src/Aspire.Hosting.Nats/NatsBuilderExtensions.cs:36]: good point. marked as obsolete. I would've thought api changes are tolerable when in preview so I'd be happy to remove if you're OK with it.

File: src/Aspire.Hosting.Nats/NatsBuilderExtensions.cs (hunk 1 above)
Comment (mitchdenny) [reply src/Aspire.Hosting.Nats/NatsBuilderExtensions.cs:57]: I think we should split out `WithJetStream` and `WithDataBindMount` and `WithDataVolume`. What I mean by that is that `WithJetStream` won't add the `-sd [path]` argument. Lets make that someting that people can do explicitly.

People might want to enable the JetStream feature set but don't care about persistence.

**Comment by mitchdenny** [comment]
I noticed that NATS doesn't have a port argument. If we were to add the port argument (as an optional) we wouldn't be able to retain the obsolete version of the method because of ambiguity. Given that NATS doesn't have a port today we should probably not add it, leave the obsolete method since there is a workaround with the `WithEndpoint(name, callback)` overload and in Aspire 9.0 remove this method.

@davidfowl seem like a reasonable plan?

File: src/Aspire.Hosting.Nats/NatsBuilderExtensions.cs (hunk 1 above)
Comment (mtmk) [reply src/Aspire.Hosting.Nats/NatsBuilderExtensions.cs:57]: where would you keep path value to be used by `WithDataBindMount` and `WithDataVolume`? Just not sure how to implement that.

**Comment by davidfowl** [comment]
Let’s add it now

**Comment by mitchdenny** [comment]
I think I was confused about the port thing. AddNats has the port argument, I thought it would have the same issue as AddSeq but AddSeq had the persistence path on that extension method instead of a seperate one like WithJetstream.

Anyway ... @mtmk I've updated your PR to split out the -sd option from WithJetstream so that they are controlled independently.

**Review by eerhardt — APPROVED** [review]

**Pull request merged**